src/core/getFrameMetadata.ts

```diff
@@ -29,14 +19,18 @@ export const getFrameMetadata = function ({
   const metadata: Record<string, string> = {
     'fc:frame': 'vNext',
   };
-  buttons.forEach((button, index) => {
-    metadata[`fc:frame:button:${index + 1}`] = button.label;
-    if (button.action) {
-      metadata[`fc:frame:button:${index + 1}:action`] = button.action;
-    }
-  });
   metadata['fc:frame:image'] = image;
-  metadata['fc:frame:post_url'] = post_url;
+  if (buttons) {
+    buttons.forEach((button, index) => {
+      metadata[`fc:frame:button:${index + 1}`] = button.label;
+      if (button.action) {
+        metadata[`fc:frame:button:${index + 1}:action`] = button.action;
+      }
+    });
+  }
+  if (post_url) {
```

Comment:
Why checking for `post_url`? The type should already check for this.

Reply:
I updated `FrameMetadata` fields. `buttons` and `post_url` are now considered optional (as they are in the Farcaster Frames API).